apis.md

```diff
@@ -9,3 +9,10 @@ The following OpenTracing APIs are currently available in the RFC stage:
 
 Please refer to the README files in the respective projects for examples of usage.
 
+### Implementation Notes
+
+OpenTracing APIs intend to express a similar experience to the user, while still being idiomatic to the language or framework. The following concepts apply to all variants.
+
+A **Span** is shared mutable state. It accumulates tags and logs, and can also create children. Finishing a span means capturing its state as a SpanRecord.
+
+A **SpanRecord** is an immutable representation of a completed span. It can be reported directly to an OpenTracing system, or encoded into a representation of an alternate system, such as HTrace or Zipkin.
```

Comment:
>
> Unfortunately, given that the API is mostly "write", there are limited
> things we can do in the tests to actually check correctness of the
> implementation. One possibility is to define some "read" API that must be
> implemented by the Reporter or some anonymous component of the tracer based
> on the data it captured, purely for compliance testing purposes.
>
ack that a read or assertion api is needed or else the tests can't really
do any correctness checking. I think a compatibility test which doesn't
check correctness isn't complete. So, basically I'd at least put a TODO on
it, if no solution is in mind, yet.